docs/contributing.rst

```diff
@@ -178,3 +183,45 @@ A much better version, free from all the cons mentioned above, is:
   - Ensure that the notebook runs when a **new** kernel is launched. Ensure that all cells run in sequential order, and that all cell outputs are evaluated.
 
 .. _existing Pull Requests: https://github.com/COSIMA/cosima-recipes/pulls
+
+
+To clone a pull request locally
++++++++++++++++++++++++++++++++
+
+If you want to test pull requests locally (i.e., to compile or run the code),
+you will need to download the pull request branch. You can do this either by cloning the branch from the pull request.
+
+In this context "locally" means somewhere you can run the code - this is probably on Gadi, but may also be on a local machine.
+
+If you are using ssh keys for command line authentication:
+
+::
+
+    git clone -b «THEIR_DEVELOPMENT_BRANCHNAME» git@github.com:«THEIR_GITHUB_USERNAME»/cosima-recipes.git
```

Comment:
```suggestion
    git clone -b «THEIR_DEVELOPMENT_BRANCHNAME» https://github.com/:«THEIR_GITHUB_USERNAME»/cosima-recipes.git
```

I would use https, it saves a lot of password typing and faff ?

Reply:
Except if you're potentially pushing changes, you'll need an SSH remote?

Reply:
Yeah. Do most reviewers push changes? They might not have permissions to push a change if its someone elses fork anyway

Reply:
> Do most reviewers push changes

Probably not! I thought it was mentioned in the instructions, but I think it was just Ed's comment about doing so. Although to be fair, people should have a good SSH key set up anyway :grin: 

> They might not have permissions to push a change if its someone elses fork anyway

If they have edit permission on the repository the PR is on, and the submitted has checked the "allow others to make changes" box (the default), they can change *just that branch* on a fork.

Reply:
Let's urge for git so reviewers can edit. Sometimes it makes the process easier...!